websites/H/HentaiVN/presence.ts

```diff
@@ -0,0 +1,567 @@
+			.querySelectorAll("span[itemprop='name']")[2]
+			.textContent.trim()} - ${document
+			.querySelectorAll("span[itemprop='name']")[3]
+			.textContent.trim()}`;
+		presenceData.smallImageKey = "doc";
+		presenceData.largeImageKey = "truyen";
+		presenceData.buttons = [
+			{
+				label: "Đọc cùng",
+				url: document.URL
+			}
+		];
+	} else if (
+		!pathname.startsWith("/forum/t") &&
+		pathname.includes("-doc-truyen-")
+	) {
+		presenceData.details = "Đang xem thông tin truyện...";
+		presenceData.state = `${document
+			.querySelectorAll("span[itemprop='name']")[2]
+			.textContent.trim()}`;
+		presenceData.buttons = [
+			{
+				label: "Xem truyện",
+				url: document.URL
+			}
+		];
+		presenceData.smallImageKey = "xem";
+		presenceData.largeImageKey = "truyen";
+	} else if (pathname.startsWith("/forum/t")) {
+		presenceData.details = "Đang xem topic ";
+		presenceData.state = `${document
+			.querySelectorAll("span[itemprop='name']")[3]
+			.textContent.trim()} - Trang ${
+			Number(searchParams.get("reply")) / 10 + 1 || "1"
+		}`;
+		presenceData.buttons = [
+			{
+				label: "Xem topic",
+				url: document.URL
+			}
+		];
+		presenceData.smallImageKey = "xem";
+		presenceData.largeImageKey = "forum";
+	} else if (pathname.startsWith("/g/")) {
+		presenceData.details = `Đang xem danh sách truyện của nhóm "${document
+			.querySelectorAll("span[itemprop='name']")[2]
+			.textContent.trim()}"`;
+		presenceData.state = `Trang ${page}`;
+		presenceData.smallImageKey = "list";
+		presenceData.largeImageKey = "truyen";
+	} else if (pathname === "/forum/") {
+		presenceData.details = "Đang duyệt forum";
+		presenceData.smallImageKey = "surf";
+		presenceData.largeImageKey = "forum";
+	} else if (pathname.startsWith("/forum/f")) {
+		presenceData.details = `Đang duyệt box ${document
+			.querySelector("b")
+			.textContent.trim()}`;
+		presenceData.state = `Trang ${
+			Number(searchParams.get("start")) / 20 + 1 || "1"
+		}`;
+		presenceData.smallImageKey = "surf";
+		presenceData.largeImageKey = "forum";
+	} else if (pathname.startsWith("/forum/c")) {
+		largebox = document.querySelector("title").textContent.trim();
+		largebox = largebox.substring(0, largebox.indexOf("-"));
+		presenceData.details = `Đang duyệt danh mục ${largebox}`;
+		presenceData.smallImageKey = "surf";
+		presenceData.largeImageKey = "forum";
+	} else if (pathname.startsWith("/forum/search-plus.php")) {
+		presenceData.details = "Đang tìm kiếm truyện nâng cao";
+		presenceData.smallImageKey = "search";
+		presenceData.largeImageKey = "home";
+		if (document.location.search.includes("search")) {
+			keystr = searchParams.getAll("tag[]");
+			namekey = `Từ khóa: ${searchParams.get("name")} - `;
+			doukey = `Doujin: ${searchParams.get("dou")} - `;
+			charkey = `Nhân vật: ${searchParams.get("char")} - `;
+			if (namekey.length === 12) namekey = "";
+
+			if (doukey.length === 11) doukey = "";
+
+			if (charkey.length === 13) charkey = "";
```

Comment:
I thought I suggested you should use ternaries here...

Reply:
done, thanks!